### ISSUE TITLE
IMPORTANT: migrate to oneVPL for newer Intel GPUs support

### ISSUE THREAD

**Issue by dmitryermilov** [Intel-Media-SDK/MediaSDK#2799]
Intel® Media SDK lifetime comes to an end in a form it exists right now. In particular,

* API 1.35 is projected to be the last API of 1.x API series
* Runtime library (libmfxhw64.so.1) is not planned to get support of new Gen platforms (ADL-S, ADL-P, and above)
* Project is going to be supported in maintenance mode, critical fixes only

All future development is planned to happen within [oneVPL](https://github.com/oneapi-src/oneVPL) library and its runtime
implementations which are direct successors of Intel® Media SDK. oneVPL introduces API 2.x series which is not backward compatible with API 1.x series (some [features got dropped](https://spec.oneapi.com/versions/latest/elements/oneVPL/source/appendix/VPL_intel_media_sdk.html)). New [VPL Runtime for Gen Graphics](https://github.com/oneapi-src/oneVPL-intel-gpu) (libmfx-gen.so.1.2) comes with the support of new Gen platforms.

### What should be packaged?

Attn. @dbermond, @jbeich, @sebastinas, @sndirsch, @tjaalton

Media SDK and oneVPL support Intel GPUs starting from gen8 BDW. However:

* New [VPL Runtime for Gen Graphics](https://github.com/oneapi-src/oneVPL-intel-gpu>) (libmfx-gen.so.1.2) supports platforms starting from Gen12 - TGL, DG1, RKL, ADL-S, ADL-P
* Media SDK runtime (libmfxhw64.so.1) supports "legacy" platforms up to (some!) Gen12 platforms - BDW, SKL, KBL, CFL, ... , TGL, DG1

MSDK and oneVPL dispatcher libraries (libmfx.so.1 or libvpl.so.2) load one of the above runtimes depending on the underlying GPU platform.

Thus, the following repos needs to be packaged to cover full range of MSDK/VPL supported platforms:
* https://github.com/oneapi-src/oneVPL (>=v2021.6.0)
* https://github.com/Intel-Media-SDK/MediaSDK (>=intel-mediasdk-21.3.4)
* https://github.com/oneapi-src/oneVPL-intel-gpu (>=intel-onevpl-21.3.4)

### Status of dependent projects

|Project|Status|
|---|---|
|ffmpeg|Not yet switched to oneVPL, patches pending package adoption by Linux distros|
|gst-plugins-bad|oneVPL support enabled|

**Comment by dbermond** [comment]
Thank you for the information. Very appreciated!

oneVPL packages on Arch Linux:

- https://aur.archlinux.org/packages/onevpl/
- https://aur.archlinux.org/packages/onevpl-cpu/
- https://aur.archlinux.org/packages/onevpl-intel-gpu/

I have placed them on the AUR (Arch User Repository) for the time being. Planning to move them to the official repositories later, as there is still low user usage and no dependent applications using them yet.

Packaging suggestions are welcome :)

**Comment by sebastinas** [comment]
I don't intend to work on packaging oneVPL for Debian.

**Comment by sndirsch** [comment]
Thanks a lot! Looks good! Everything addressed from our side! :

Stefan

**Comment by sndirsch** [comment]
Hello again. Unfortunately I found a similar issue. This time in
oneapi-src/oneVPL sources. It seems the component contains code licensed
under "MS Vistual Studio" licsense. AFAIK this is incompatible
to GPL-3.0 license.i.e. you can't link GPL 3.0 sources against libvpl. This
makes it difficult for a Linux distributor to ship this library. :-( Is
this an oversight, intentional or just can't be changed?
I've opened an issue for this, but haven't seen any response to this up to
now. :-(

https://github.com/oneapi-src/oneVPL/issues/27

Thanks,
Stefan

**Comment by eero-t** [comment]
Summarizing here....

> Unfortunately I found a similar issue. This time in oneapi-src/oneVPL sources. It seems the component contains code licensed under "MS Vistual Studio" license. AFAIK this is incompatible to GPL-3.0 license.i.e. you can't link GPL 3.0 sources against libvpl. This makes it difficult for a Linux distributor to ship this library. :-( Is this an oversight, intentional or just can't be changed?

According to https://github.com/oneapi-src/oneVPL/issues/27 this was just a misunderstanding (that license applies only to Windows builds which are not distributed under GPL) which is / will be cleared in the documentation.

Additionally, "oneVPL-intel-gpu" (Intel GPU backend for oneVPL frontend) project build issue with GCC v11.2 (used in Ubuntu 21.10 and 22.04) is now fixed, but there is not yet a tagged version with that fix.

**Comment by dvrogozh** [comment]
> project build issue with GCC v11.2 is now fixed, but there is not yet a tagged version with that fix

Yes, fixed in https://github.com/oneapi-src/oneVPL-intel-gpu/commit/a8702a9eb94a306ec8f592b5fcac41c84564f7c4. Will go into one of the next release or release updates.

**Comment by nyanmisaka** [comment]
MSDK dispatcher can't load `libmfx64-gen.dll` on Windows. https://github.com/Intel-Media-SDK/MediaSDK/issues/2904
No forward compatibility like the behavior on Linux.
